scripts/staging/fm-binclass.dml

```diff
@@ -0,0 +1,173 @@
+    [mV, vV]   = adam::init(V);
+    
+    # Regularization
+    lambda = 5e-04
+    
+    # Optimize
+    print("Starting optimization")
+    batch_size = 64
+    iters = ceil(N / batch_size)
+    for (e in 1:epochs) {
+      for (i in 1:iters) {
+        # Get the next batch
+        beg = ((i-1) * batch_size) %% N + 1
+        end = min(N, beg + batch_size - 1)
+        X_batch = X[beg:end,]
+        y_batch = y[beg:end,]
+    
+        # 3.Send inputs through fm::forward
+        y = fm::forward(X_batch, w0, W, V);  
+    
+        # 4.Send the above result through sigmoid::forward
+        sfy = sigmoid::forward(y);
+    
+        # 5.Send the above result through log_loss::forward
+        loss = log_loss::forward(sfy);
+        
+        # Compute loss & accuracy for training & validation data every 100 iterations.
+        if (i %% 100 == 0) {
+          # Compute training loss & accuracy
+          loss_data   = cross_entropy_loss::forward(loss, y_batch);
+          loss_reg_X  = l2_reg::forward(X, lambda); 
+          loss_reg_w0 = l2_reg::forward(w0, lambda);
+          loss_reg_W  = l2_reg::forward(W, lambda);
+          loss_reg_V  = l2_reg::forward(V, lambda);
+          
+          accuracy = (rowIndexMax(loss) == rowIndexMax(y_batch));
+          loss = loss_data + loss_reg_X + loss_reg_w0 + loss_reg_W + loss_reg_V;
+          
+          # Compute validation loss & accuracy
+          probs_val = predict(X_val)
+          loss_val = cross_entropy_loss::forward(probs_val, y_val)
+          accuracy_val = mean(rowIndexMax(probs) == rowIndexMax(y_val))
+
+          # Output results
+          print("Epoch: " + e + ", Iter: " + i + ", Train Loss: " + loss + ", Train Accuracy: "
+                + accuracy + ", Val Loss: " + loss_val + ", Val Accuracy: " + accuracy_val)
+        }
+        
+        # 6.Send the result of sigmoid::forward and the correct labels y to log_loss::backward
+        dsfy = log_loss::backward(sfy, y);
+    
+        # 7.Send the above result through sigmoid::backward
+        dy = sigmoid::backward(dsfy, y);
+    
+        # 8.Send the above result through fm::backward
+        [dX, dw0, dW, dV] = fm::backward(dy, X, w0, W, V);
+        
+        # 9.Call adam::update for all parameters
+                
+        #X = adam::update(X, dX, lr, beta1, beta2, epsilon, t, mX, vX);
+        w0 = adam::update(w0, dw0, lr, beta1, beta2, epsilon, t, mw0, vw0);
+        W  = adam::update(W, dW, lr, beta1, beta2, epsilon, t, mW, vW );
+        V  = adam::update(V, dV, lr, beta1, beta2, epsilon, t, mV, vV );
+      }
+    }
+}
+
+
+predict = function(matrix[double] X)
+    return (matrix[double] out) {
+  /*
+   * Computes the predictions for the given inputs.
+   *
+   * Inputs:
+   *  - X : n examples with d features, of shape (n, d).
+   *  - w0, W, V : trained model parameters. 
+   *  
+   * Outputs:
+   *  - out : target vector, y.
+   */
+    
+    # 1.initialize fm core
+    [w0, W, V] = fm::init(d, k);
```

Comment:
Pass `w0, W, V` into the predict function so that you can make predictions with trained models, rather than create ones from scratch here.